app/controllers/rules_of_origin/steps_controller.rb

```diff
@@ -26,26 +26,31 @@ def show
     private
 
     def wizard_store_key
-      :rules_of_origin
+      "rules_of_origin-#{params['commodity']}-#{params['country']}"
     end
 
     def step_path(step_id = params[:id], ...)
-      rules_of_origin_step_path(step_id, ...)
+      rules_of_origin_step_path(params['commodity'],
+                                params['country'],
+                                step_id,
+                                ...)
     end
 
     def check_session_data
-      if service_name != wizard_store['service']
+      if service_name != wizard_store['service'] ||
+          params['commodity'].blank? ||
+          params['country'].blank?
 
         redirect_to return_to_commodity_path
         wizard_store.purge!
       end
     end
 
     def return_to_commodity_path
-      return find_commodity_path if wizard_store['commodity_code'].blank?
+      return find_commodity_path if params['commodity'].blank?
```

Comment:
Are we passing an object as a param? commodity code/goods nomenclature code is more explicit